staging/src/k8s.io/client-go/tools/events/helpers.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/api/events/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	typedV1beta1 "k8s.io/client-go/kubernetes/typed/events/v1beta1"
+)
+
+// SearchEvents finds events about the specified object.
+func SearchEvents(client typedV1beta1.EventInterface, ref *v1.ObjectReference) (*v1beta1.EventList, error) {
+	stringRefKind := string(ref.Kind)
+	var refKind *string
+	if len(stringRefKind) > 0 {
+		refKind = &stringRefKind
+	}
+	stringRefUID := string(ref.UID)
+	var refUID *string
+	if len(stringRefUID) > 0 {
+		refUID = &stringRefUID
+	}
+	fieldSelector := getFieldSelector(&ref.Name, &ref.Namespace, refKind, refUID)
+	return client.List(metav1.ListOptions{FieldSelector: fieldSelector.String()})
```

Comment:
there are a lot of other list options this method masks (ability to watch, timeout, page/continue, to request a minimum resource version, specify a label selector, etc).

would it make more sense to just expose the helper that constructs a v1beta1 field selector?

something like
```go
func GetFieldSelector(groupVersion schema.GroupVersion, name, namespace, kind, uid string) (fields.Selector, error) {
...
}
```

that would let us implement the selector for `v1` and `events.k8s.io/v1beta1` (and future versions), and would let the caller use the returned selector for any list/watch purpose they wanted

Reply:
Sure, make sense, I will update.